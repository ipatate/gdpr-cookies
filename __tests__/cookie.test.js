import Cookie from '../src/utils/Cookie';

test('cookie get all value', () => {
  const _cookie = new Cookie();
  _cookie.set('plop', 'ok');
  const all = _cookie.getAll();
  expect(all).toMatchSnapshot();
});

test('cookie set value', () => {
  const _cookie = new Cookie();
  _cookie.set('plop', 'ok');
  const all = _cookie.getAll();
  expect(all).toMatchSnapshot();
});

test('cookie set value not valid', () => {
  const _cookie = new Cookie();
  _cookie.set('plop', []);
  const all = _cookie.get('plop');
  expect(all).toBe('[]');
});

test('cookie get single value', () => {
  const _cookie = new Cookie();
  _cookie.set('lol', 'ok');
  const all = _cookie.get('lol');
  expect(all).toBe('ok');
});

test('cookie get single value not exist', () => {
  const _cookie = new Cookie();
  const all = _cookie.get('foo');
  expect(all).toBeUndefined();
});

test('cookie remove single value', () => {
  const _cookie = new Cookie();
  _cookie.set('lol', 'ok');
  const all = _cookie.remove('lol');
  expect(all).toBeUndefined();
});
