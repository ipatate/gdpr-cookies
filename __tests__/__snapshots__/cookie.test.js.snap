// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`cookie get all value 1`] = `
Object {
  "plop": "ok",
}
`;

exports[`cookie set value 1`] = `
Object {
  "plop": "ok",
}
`;
